samcli/lib/cookiecutter/interactive_flow.py

```diff
@@ -49,14 +56,22 @@ def run(self, context: Dict) -> Dict:
         ----------
         context: Dict
             The cookiecutter context before prompting this flow's questions
+        preload_values_toml_file_path: Path
+            The toml file path to preload values from
 
-        Returns: A new copy of the context with user's answers added to the copy such that each answer is
-                 associated to the key of the corresponding question
+        Returns
+        -------
+        A new copy of the context with user's answers added to the copy such that each answer is
+             associated to the key of the corresponding question
         """
-        context = context.copy()
+        context = (
+            preload_values_from_toml_file(preload_values_toml_file_path, context)
+            if preload_values_toml_file_path
+            else context.copy()
+        )
```

Comment:
The design was meant to be generic; It allows that the cookiecutter template to have multiple interactive-flows which run back to back and merge all the responses together to create the cookiecutter context. Every interactive-flow accepts the context of the previous one, copy it, add its additional context and return the new one.

Now, with the preload_values_from_toml_file, the context of the previous interactive-flows will be lost.

Reply:
updated as discussed offline